.circleci/config/commands/go_test.yml

```diff
@@ -156,6 +142,20 @@ steps:
           mkdir workspace
           echo ${CONTAINER_ID} > workspace/container_id
 
+          # Hack: Docker permissions appear to have changed; let's explicitly
+          # add a new group with the correct host gid to the docker container
+          # for safety, and then add the user to the (potentially new) group.
+          #
+          # Notably, in this shell pipeline we see:
+          # uid=1001(circleci) gid=1002(circleci) groups=1002(circleci)
+          #
+          # but inside the docker image below, we see:
+          # uid=3434(circleci) gid=3434(circleci) groups=3434(circleci)
+          #
+          # See also: https://github.com/CircleCI-Public/cimg-base/issues/122
+          HOST_GID="$(id -g)"
+          docker exec ${CONTAINER_ID} sh -c 'groupadd hcp-circleci -g '"$HOST_GID"' ; usermod $UID -a -G '"$HOST_GID"
```

Comment:
Quoting gets a little tricky, but the UID we want is inside the container, hence why its in single quotes, but the GID needs to be passed through from this shell pipeline.

Reply:
You could use -e HOST_GID to avoid the tricky quotes.

Reply:
Ah, makes sense `exec` would support that I suppose, good idea!